### PR TITLE
Populate the "Attributes" section based on OAS data

### PR DESCRIPTION
**Pull request by pdwilson12**
Bug/issue #, if applicable: rdar://121475839&121224338

## Summary

Important details recorded in symbol graphs about the allowed values of a data type, such as minimum and maximum values, default values, and enumerated values, were not getting used to generate an `AttributeRenderSection` on a symbol's page. That data now gets passed through to the render JSON.

## Dependencies

None.

## Testing

Add one of the value constraint fields to a data type in a symbol graph (eg, `"minimum": 12`) and compile the documentation. The render JSON for the data type should include an "attributes" primary content section with the attribute value.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary (N/A)

### CODE REVIEW

File: Tests/SwiftDocCTests/Test Bundles/DictionaryData.docc/dictionary.symbols.json [hunk 1]
```diff
@@ -239,6 +239,32 @@
     },
     {
       "accessLevel": "public",
+      "allowedValues": [
+        "Classic Rock",
+        "Folk",
+        null
```
Comment (d-ronnqvist) [inline Tests/SwiftDocCTests/Test Bundles/DictionaryData.docc/dictionary.symbols.json:245]: Question: Is this decoded as a null value in the JSON or as a string? I see that the test expects it as a string. How is this expected to be displayed on the page?

File: Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/AttributesSectionTranslator.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                        return RenderAttribute.maximumLength(String(value))
+                    case (.default, let value as SymbolGraph.AnyScalar):
+                        return RenderAttribute.default(String(value))
+                    case (.allowedTypes, let types as [SymbolGraph.Symbol.TypeDetail]):
+                        let tokens = types.compactMap {
+                            if let fragments = $0.fragments {
+                                return translateFragments(fragments)
+                            } else {
+                                return nil
+                            }
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/AttributesSectionTranslator.swift:68]: Minor: This is a good use of `Optional.map`

```suggestion
                            $0.fragments.map(translateFragments)
```

File: Tests/SwiftDocCTests/Test Bundles/DictionaryData.docc/dictionary.symbols.json (hunk 1 above)
Comment (pdwilson12) [reply Tests/SwiftDocCTests/Test Bundles/DictionaryData.docc/dictionary.symbols.json:245]: JSON allows proper null/nil values and that is what is being indicated here. Internally it stays flagged as `null` up until the render JSON phase where it is converted to a string for display. There isn't an obvious way to flag that in the rendered page unless we force all strings to be quoted, which isn't ideal. It is up to the reader to recognize that and entry that reads `null` means the nil value and not the string "null".

File: Sources/SwiftDocC/Semantics/Symbol/Symbol.swift [hunk 3]
```diff
@@ -536,6 +565,12 @@ extension Symbol {
         set { declarationVariants.firstValue = newValue }
     }
     
+    /// The first variant of the symbol's attributes.
+    public var attributes: [RenderAttribute.Kind: Any]? {
+        get { attributesVariants.firstValue! }
+        set { attributesVariants.firstValue = newValue }
+    }
+
```
[the remark this answers is not in the document]
Comment (pdwilson12) [reply Sources/SwiftDocC/Semantics/Symbol/Symbol.swift]: Deleted

File: Sources/SwiftDocC/Semantics/Symbol/Symbol.swift [hunk 4]
```diff
@@ -304,6 +308,7 @@ public final class Symbol: Semantic, Abstracted, Redirected, AutomaticTaskGroups
         self.mixinsVariants = mixinsVariants
         
         for (trait, variant) in mixinsVariants.allValues {
+            var attributesVariants: [RenderAttribute.Kind: Any] = [:]
```
[the remark this answers is not in the document]
Comment (pdwilson12) [reply Sources/SwiftDocC/Semantics/Symbol/Symbol.swift]: Good point. Updated.

File: Sources/SwiftDocC/Semantics/Symbol/Symbol.swift [hunk 5]
```diff
@@ -317,9 +322,33 @@ public final class Symbol: Semantic, Abstracted, Redirected, AutomaticTaskGroups
...
+                    attributesVariants[.default] = attribute.value
+                
+                case let attribute as SymbolGraph.Symbol.TypeDetails:
+                    attributesVariants[.allowedTypes] = attribute.value
+                case let attribute as SymbolGraph.Symbol.AllowedValues:
+                    attributesVariants[.allowedValues] = attribute.value
                 default: break;
                 }
             }
+            if attributesVariants.count > 0 {
```
[the remark this answers is not in the document]
Comment (pdwilson12) [reply Sources/SwiftDocC/Semantics/Symbol/Symbol.swift]: Updated to `isEmpty`

**Review by d-ronnqvist — APPROVED** [review]

**Comment by pdwilson12** [comment]
@swift-ci please test

**Comment by pdwilson12** [comment]
@swift-ci please test

**Comment by pdwilson12** [comment]
@swift-ci please test

**Comment by pdwilson12** [comment]
@swift-ci please test

**Comment by pdwilson12** [comment]
@swift-ci please test

**Pull request merged**